### PR TITLE
drivers: sensors: icm42605: use gpio_dt_spec and fix trigger bug

### PR DESCRIPTION
**Pull request by aunsbjerg**
Converts to using gpio_dt_spec for interrupt gpio and fixes a exception-causing bug in passing data to the thread from the thread creation routine in the trigger init.

### CODE REVIEW

File: drivers/sensor/icm42605/icm42605_trigger.c [hunk 1]
```diff
@@ -86,16 +86,11 @@ static void icm42605_thread_cb(const struct device *dev)
 	gpio_pin_interrupt_configure_dt(&cfg->gpio_int, GPIO_INT_EDGE_TO_ACTIVE);
 }
 
-static void icm42605_thread(int dev_ptr, int unused)
+static void icm42605_thread(struct icm42605_data *drv_data)
```
[the remark this answers is not in the document]
Comment (aunsbjerg) [reply drivers/sensor/icm42605/icm42605_trigger.c:89]: Yes. There are multiple examples of doing what I do here, see `adxl362_trigger.c:60`, but if that is no an acceptable approach, I can of course change to the standard thread signature.

[the remark this answers is not in the document]
Comment (Abramo-Bagnara) [reply drivers/sensor/icm42605/icm42605_trigger.c:89]: My suggestion is to use always the fixed signature and do the needed conversion inside the body of the specific thread function.
This gives a strong warranty that ABI for argument passing is not violated.

**Review by gmarull — DISMISSED** [review]

**Review by teburd — DISMISSED** [review]

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Comment by aunsbjerg** [comment]
Removed unrelated formatting changes and added a commit adding ICM42605 w. trigger support to build_all test suite.

**Review by gmarull — DISMISSED** [review]

**Comment by aunsbjerg** [comment]
Fixed CI failure because missing signoff message

**Review by MaureenHelm — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Pull request merged**